parl/framework/model_base.py

```diff
@@ -143,6 +143,83 @@ def set_params(self, params, gpu_id):
         for (param_name, param) in list(zip(self.parameter_names, params)):
             set_value(param_name, param, gpu_id)
 
+    def _get_parameter_names(self, obj):
+        """ Recursively get parameter names in obj,
+            mainly used to get parameter names of a parl.Network.
+            The order of parameter names will be consistent between
+            different instances of same parl.Network.
+
+        Args:
+            obj: parl.Network/parl.LayerFunc/list/tuple/set/dict
+
+        Returns:
+            parameter_names: list of string, all parameter names in obj
+        """
+
+        parameter_names = []
+        for attr in sorted(obj.__dict__.keys()):
+            val = getattr(obj, attr)
+            if isinstance(val, Network):
+                parameter_names.extend(self._get_parameter_names(val))
+            elif isinstance(val, LayerFunc):
+                for attr in val.attr_holder.sorted():
+                    if attr:
+                        parameter_names.append(attr.name)
+            elif isinstance(val, tuple) or isinstance(val, list):
+                for x in val:
+                    parameter_names.extend(self._get_parameter_names(x))
+            elif isinstance(val, dict):
+                for x in list(val.values()):
+                    parameter_names.extend(self._get_parameter_names(x))
+            else:
+                # for any other type, won't be handled. Eg: set
+                pass
+        return parameter_names
+
+    def _get_parameter_pairs(self, src, target):
+        """ Recursively get pairs of parameter names between src and target
+
+        Args:
+            src: parl.Network/parl.LayerFunc/list/tuple/set/dict
```

Comment:
When writing comments for Args, we usually use the following format:
src(type): explaination about what src is here

Reply:
done